### PR TITLE
Add Extension capability for handling custom data types

### PR DESCRIPTION
**Pull request by niahoo**
Hello, 

I made a small modification so I an extend your class and pass any data I want to the renderer. I must handle document objects, and implement `ArrayAccess` is not enough to have correct data :

```PHP

class DocumentTable extends CliTable {

  protected function getValue($doc, string $key)
  {
    return $doc->getMetadata($key, null);
  }

}

```

So maybe you want to merge it :)

I also fixed a bug when displaying data with UTF-8 characters, but maybe that would require to add symfony's mbstring polyfill as a dependency.

### CODE REVIEW

File: src/jc21/CliTable.php [hunk 1]
```diff
@@ -400,7 +400,7 @@ protected function getFormattedRow($rowData, $columnLengths, $header = false) {
                 $color = $this->fields[$key]['color'];
             }
 
-            $fieldLength  = strlen($field) + 1;
+            $fieldLength  = mb_strlen($field) + 1;
```
Comment (peter279k) [inline src/jc21/CliTable.php]: If we let the `strlen` replace with `mb_strlen`, we have to define the `ext-mbstring` inside `require` block in `composer.json`.

The sample definition is as follows:

```
......
 "require": {
        "php": ">=5.3.0",
        "ext-mbstring": "*"
},
......
```

File: src/jc21/CliTable.php [hunk 2]
```diff
@@ -531,6 +531,21 @@ protected function getColorFromName($colorName)
...
+     *
+     * @access protected
+     * @param  (array | ArrayAccess)  $row
+     * @param  string  $key
+     * @param  any  $rowKey
+     * @return string
+     */
+    protected function getValue($row, string $key, $rowKey)
+    {
+        return $row[$key];
```
Comment (peter279k) [inline src/jc21/CliTable.php]: According to this `getValue` method, it looks like the `$rowKey` parameter is not used.

Please remove this.

Comment (jc21) [reply src/jc21/CliTable.php]: I think that might be on purpose, as with extended classes the function signature needs to be the same in php 7+ versions (can't remember which one specifically)

Comment (niahoo) [reply src/jc21/CliTable.php]: Hello, 

Yes that was the point : check the method signature and see what you will receive as args if you override the method.

**Comment by niahoo** [comment]
So, sorry for the long time, new job here.

I checked the code and I installed another library for my CLI display. So maybe close this or do what you want but as far as I am concerned I will not put more time in this ; I'm sorry.

**Pull request closed without merging**